app/models/early_career_teacher_profile_declaration.rb

```diff
@@ -0,0 +1,6 @@
+# frozen_string_literal: true
+
+class EarlyCareerTeacherProfileDeclaration < ApplicationRecord
+  belongs_to :early_career_teacher_profile
+  include Declarable
+end
```

Comment:
What is the job of this class? As I see it, if we inlined `Declarable`, we would get

```
class EarlyCareerTeacherProfileDeclaration < ApplicationRecord
  belongs_to :early_career_teacher_profile
  has_one :profile_declaration, as: :declarable, touch: true
end
```

It looks a lot like a middle man without any other function than being a middle man. I guess it keeps the time of when it was created, but so does `ProfileDeclaration`. 

Why would we want to do it this way, rather than by linking directly to the relevant profiles?

Reply:
It is exactly that. A proxy for the profile. Part of the reason was to not have polymorphic foreign keys to different profiles types directly. This mirrors the current profile layout and will continue to do so later. If, and when, the profiles collapse into the `participant_profiles`, the `participant_profile_declarations` will follow suit. (Especially if npq_profiles also becomes linked in that way.)